hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -895,7 +893,7 @@ private Set<String> getMetadataPartitionsToUpdate() {
     return getEnabledPartitionTypes().stream().map(MetadataPartitionType::getPartitionPath).collect(Collectors.toSet());
   }
 
-  public void buildMetadataPartitions(HoodieEngineContext engineContext, List<HoodieIndexPartitionInfo> indexPartitionInfos) throws IOException {
+  public void buildMetadataPartitions(HoodieEngineContext engineContext, List<HoodieIndexPartitionInfo> indexPartitionInfos, String instantTime) throws IOException {
```

Comment:
Did we walk through the commit time logic used to initialized new partitions and ensured new logic is intact. 

for eg: 
in 0.15.x:
say timeline was 
t1, t2, .... t40, t41, t42.inflight, t43,... t50. 

instant time of indexing commit in data table will be t50. but t43_004 will be the commit to instatiate say col stats asynchronously. 

after this patch, we might be using t50. but isn't t50.dc already completed in MDT to apply changes for t50.dc from data table. 
don't we need to generate another commit time unique to MDT timeline for async initialization purpose

Reply:
lets also check the logic when N no of MDT partitions are initialized async.

Reply:
> t50. but isn't t50.dc already completed in MDT to apply changes for t50.dc from data table.

Not sure I got your point, why the dc and index commit share the same instant time?